Crash: Addition expression: _Complex + integer

```c
void foo(void) {
    (_Complex double)1 + 2;
}
```

```
thread 31081065 panic: reached unreachable code
/Users/ehaas/source/arocc/src/Parser.zig:4077:29: 0x105ff7946 in Parser.Result.add (arocc)
                    else => unreachable,
                            ^
/Users/ehaas/source/arocc/src/Parser.zig:4560:28: 0x105ff68be in Parser.addExpr (arocc)
                try lhs.add(plus.?, rhs, p);
                           ^
/Users/ehaas/source/arocc/src/Parser.zig:4517:28: 0x105ff58f4 in Parser.shiftExpr (arocc)
    var lhs = try p.addExpr();
                           ^
```

Further perusal of the C11 standard shows me that the above only works because `_Static_assert` actually requires that all the elements of the condition be integer constants; clang actually does evaluate constant expressions with e.g. complex numbers:

```c
void foo(void) {
    const int x = (1.0f + (_Complex long double)1.0fi) - (1.0f + (_Complex long double)1.0fi);
    const int y = 1/x;
}
```
```
test.c:3:20: warning: division by zero is undefined [-Wdivision-by-zero]
    const int y = 1/x;
                   ^~
```

So I guess we need to add some entries to the `val` enum in `Result` - pointers to `f128` and `std.math.Complex(f128)` perhaps?

--- a/src/Parser.zig
+++ b/src/Parser.zig
@@ -4801,7 +4801,9 @@ fn castExpr(p: *Parser) Error!Result {
                     try p.errStr(.invalid_cast_to_pointer, l_paren, try p.typeStr(operand.ty));
 
                 const is_unsigned = ty.isUnsignedInt(p.pp.comp);
-                if (is_unsigned and operand.val == .signed) {
+                if (!ty.isInt()) {
+                    try operand.saveValue(p);
+                } else if (is_unsigned and operand.val == .signed) {
                     const copy = operand.val.signed;
                     operand.val = .{ .unsigned = @bitCast(u64, copy) };
                 } else if (!is_unsigned and operand.val == .unsigned) {
@@ -4948,6 +4950,7 @@ fn unExpr(p: *Parser) Error!Result {
                 .specifier = .pointer,
                 .data = .{ .sub_type = elem_ty },
             };
+            try operand.saveValue(p);
             try operand.un(p, .addr_of_expr);
             return operand;
         },

--- a/test/cases/unavailable results.c
+++ b/test/cases/unavailable results.c
@@ -1,0 +1,12 @@
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-value"
+void foo(void) {
+    1.0f == 1.0f;
+    (double)1 + 1U == 2.0;
+    (_Complex float)1 + 2U;
+    1U + (int *)1;
+    1U + &2;
+}
+#pragma GCC diagnostic pop
+
+#define EXPECTED_ERRORS "unavailable results.c:8:10: error: cannot take the address of an rvalue" \

